### ISSUE TITLE
[REVIEW]: SAILS: Spectral Analysis In Linear Systems

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1982]
**Submitting author:** @ajquinn (<a href="http://orcid.org/0000-0003-2267-9897">Andrew Quinn</a>)
**Repository:** <a href="https://vcs.ynic.york.ac.uk/analysis/sails" target ="_blank">https://vcs.ynic.york.ac.uk/analysis/sails</a>
**Version:** v1.0.0
**Editor:** @mjsottile
**Reviewers:** @szorowi1, @agricolab
**Archive:** <a href="https://doi.org/10.5281/zenodo.3698334" target="_blank">10.5281/zenodo.3698334</a>

## Status

[![status](https://joss.theoj.org/papers/c4e1086aa6f5568089735cc487111d89/status.svg)](https://joss.theoj.org/papers/c4e1086aa6f5568089735cc487111d89)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/c4e1086aa6f5568089735cc487111d89"><img src="https://joss.theoj.org/papers/c4e1086aa6f5568089735cc487111d89/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/c4e1086aa6f5568089735cc487111d89/status.svg)](https://joss.theoj.org/papers/c4e1086aa6f5568089735cc487111d89)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="vcs.ynic.york.ac.uk/analysis/sails" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@szorowi1 & @mnarayan, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @mjsottile know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="vcs.ynic.york.ac.uk/analysis/sails">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@ajquinn) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="vcs.ynic.york.ac.uk/analysis/sails">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@ajquinn) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @szorowi1, @mnarayan it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1982 with the following error: 

 Can't find any papers to compile :-(

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1145/382043.382304 is OK
- 10.1093/biomet/37.3-4.409 is OK
- 10.1007/s004229900137 is OK
- 10.1101/gr.092759.109 is OK
- 10.3389/fnhum.2014.00448 is OK
- 10.1007/bf00198091 is OK
- 10.1007/pl00007990 is OK

MISSING DOIs

- https://doi.org/10.1007/3-540-27752-8 may be missing for title: New Introduction to Multiple Time Series Analysis

INVALID DOIs

- None
```

**Comment by mjsottile** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01982/joss.01982/10.21105.joss.01982.pdf)

**Comment by szorowi1** [comment]
Hi @mjsottile, I'm beginning my review. 

One immediate problem I'm running into is that I cannot seem to log into the York NeuroImaging Centre Gitlab Service in order to create a new issue in the target repository. (I'm having no trouble logging into Gitlab in general.) Maybe @mnarayan is having better luck?

**Comment by mjsottile** [comment]
Hi @szorowi1, thanks for pointing this out.  I also checked and wasn't able to make an account myself to post issues.  @AJQuinn - do you have any input on how we can submit issues for the project as part of the review process?

This is not urgent - I understand that many people are off for the end of year holidays, and I don't want this to cause unnecessary work right now.  I'll followup around the new year.

**Comment by AJQuinn** [comment]
Hello - thanks all for your time with our submission.

Apologies for the issue board issue, this was a bit more complicated on a self-hosted gitlab than we expected. We'll make it public ASAP and update here once it is ready.

Please let me know if there are any other issues in the meantime.

Cheers,
Andrew

**Comment by AJQuinn** [comment]
Hello, we've fixed the issue and the repo is now open for external users to post issues etc. There are now also instructions and details on contributing in the top-level README. 

We've tested this ourselves, so should be working for anyone. Please let me know if there are any further problems.

Thanks again for your time.

Cheers
Andrew

**Comment by mjsottile** [comment]
@szorowi1 I believe the problem you encountered regarding submitting issues has been addressed.  Please let me know if you or @mnarayan run into any further issues completing the review.

**Comment by szorowi1** [comment]
Happy new year all! To confirm, I am now successfully able to submit issues/PRs in the target repository. 

Today I've continued my review. In brief, I've completed the following:
- Downloaded and installed the package.
- Recreated the tutorials on my personal computer.
- Tested out core functions on independently simulated and actual time-series data (my own EEG data).
- Reviewed the JOSS paper.
- Submitted several minor issues. 

My only major note at this time is that the API documentation for `sails` could be improved. I've opened a corresponding issue in the target repository, and will wait to hear back from the authors. Documentation aside, seems like a cool package!

**Comment by mnarayan** [comment]
@mjsottile It looks like I am unable to check off boxes on my checklist. This has never happened before?

**Comment by arfon** [comment]
@mnarayan - you need to accept this invite before you can work on the checklist: https://github.com/openjournals/joss-reviews/invitations

**Comment by AJQuinn** [comment]
Hello, we've completed quite a big set of changes (mostly docs) as rightly suggested by @szorowi1. We've closed several smaller issues but have left the one with the big changes to the doc open for now in case anything else comes up (https://vcs.ynic.york.ac.uk/analysis/sails/issues/32; https://vcs.ynic.york.ac.uk/analysis/sails/merge_requests/30). You can view the docs build for that branch here( https://sails.readthedocs.io/en/32-joss-review-documentation/index.html). 

@szorowi1, please let us know if there is any other feedback or changes to be made!

Thanks all,
Andrew

**Comment by szorowi1** [comment]
To update: just had the chance to look through the updated docs and I think they look great! @AJQuinn  mentioned there may be more restructuring to their API documentation, so I'll wait to see if that gets implemented. Otherwise, I'm thinking things are looking very good!

**Comment by mjsottile** [comment]
@mnarayan It looks like the issue with the checklist should be resolved - are you able to proceed with the review?

**Comment by mjsottile** [comment]
To update, I'm going to seek out an additional reviewer since @mnarayan seems to have gone silent.  I will try to find this additional reviewer soon so the review doesn't stall much longer.

**Comment by AJQuinn** [comment]
OK, thank you - please let me know if there is anything I can do to help out.

**Comment by mjsottile** [comment]
:wave: @agricolab Would you be available to join this review?  One of the existing reviewers went silent and I recall that you said you were busy through January when we initiated the review in December.  Do you have time in your schedule now that January has ended?  If not, no problem - I'll continue searching.

**Comment by agricolab** [comment]
can do, @mjsottile.

**Comment by mjsottile** [comment]
@whedon add @agricolab as reviewer

**Comment by whedon** [comment]
OK, @agricolab is now a reviewer

**Comment by mjsottile** [comment]
@whedon remove @mnarayan as reviewer

**Comment by whedon** [comment]
OK, @mnarayan is no longer a reviewer

**Comment by mjsottile** [comment]
@whedon remind @agricolab in 1 week

**Comment by whedon** [comment]
Reminder set for @agricolab in 1 week

**Comment by agricolab** [comment]
I started the review, but it is a quite extensive package. I will need a couple of days.

**Comment by agricolab** [comment]
I added some issue when running the tutorials to the projects repository, mostly minor bugs or warnings. The package is powerful and relatively well-documented. Yet, it is also complex, and i feel that there is some room for improving documentation to make it more applicable.

I have the impression that the paper is well-written. But i feel that a
- [ ] short paragraph describing the advantages of using sails compared to e.g. MNE or ConnectiviPy might be useful.
- [ ] short paragraph on target audience (e.g. Control engineer, Neuroscientist, Economist, etc.) and one or two use cases why they should use `sails`

**Comment by whedon** [comment]
:wave: @agricolab, please update us on how your review is going.

**Comment by agricolab** [comment]
Considering the eventual review & merge of the pull requests, i have the impression that all of my comments have been addressed. Thanks for the effort, and a big hooray to the authors for creating such a toolbox. I can certainly see myself using it and recommending it to my colleagues and students.

**Comment by AJQuinn** [comment]
Thank you for the review @agricolab, we've completed the review and merged all the relevant changes to your issues. The master branch at https://vcs.ynic.york.ac.uk/analysis/sails now has all the changes relating to [joss review] issues from both reviewers merged in.

**Comment by AJQuinn** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01982/joss.01982/10.21105.joss.01982.pdf)

**Comment by mjsottile** [comment]
Excellent - I will take a look through everything in the next couple days to move the review on to the next stage.

**Comment by mjsottile** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1145/382043.382304 is OK
- 10.1093/biomet/37.3-4.409 is OK
- 10.1007/s004229900137 is OK
- 10.1101/gr.092759.109 is OK
- 10.3389/fnhum.2014.00448 is OK
- 10.1007/bf00198091 is OK
- 10.1007/pl00007990 is OK

MISSING DOIs

- https://doi.org/10.1007/3-540-27752-8 may be missing for title: New Introduction to Multiple Time Series Analysis

INVALID DOIs

- None
```

**Comment by mjsottile** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01982/joss.01982/10.21105.joss.01982.pdf)

**Comment by mjsottile** [comment]
@AJQuinn I just went through everything and we're getting close.  A few items:

- Please add the DOI for the book “New Introduction to Multiple Time Series Analysis” (see: https://link.springer.com/book/10.1007/978-3-540-27752-1#about).  
- In the section “state of the field”.  The first sentence is worded awkwardly.  I’d change the first sentence to something similar to “Several Python packages provide related functionality to SAILS”.  

Once that has been addressed, please make a tagged release, archive it on Zenodo or figshare, and post the DOI and version in this thread.  Please ensure that the metadata for the archive matches the information in the paper (e.g., author list and title).

At that point we should be good to pass the submission to the EIC to finalize the process.

**Comment by AJQuinn** [comment]
Thank you @mjsottile, great that we're getting there! I've fixed the DOI in a branch (https://vcs.ynic.york.ac.uk/analysis/sails/-/merge_requests/55) but I'm afraid I need to double check the second point... 

The  'State of the field' section currently does start with "Several Python packages provide related functionality to SAILS.". I'm happy to clarify the any unclear writing - but not sure if this is the right section or the right sentence. 

Could you clarify which sentence to update? apologies if I'm missing something obvious.  I'm ready to move ahead with the deposit otherwise.

Cheers,
Andrew

**Comment by mjsottile** [comment]
Sorry about that - I believe I was working off of a PDF that was generated before the changes that you made earlier this week on March 2nd.  Looking at the re-generated PDF it looks all good.  Go ahead and proceed with the deposit.

**Comment by szorowi1** [comment]
Congrats @AJQuinn and thanks for the positive reviewing experience!

**Comment by mjsottile** [comment]
@whedon set 10.5281/zenodo.3698334 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3698334" target="_blank">10.5281/zenodo.3698334</a> is the archive.

**Comment by mjsottile** [comment]
@whedon set v1.0.0 as version

**Comment by whedon** [comment]
OK. v1.0.0 is the version.

**Comment by mjsottile** [comment]
Looks like we're good to go - I'll hand it off to the EICs to finalize the process.  Thanks @szorowi1 and @agricolab for your work on the reviews, and @AJQuinn for your work on your paper and software!

**Comment by mjsottile** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1145/382043.382304 is OK
- 10.1093/biomet/37.3-4.409 is OK
- 10.1007/s004229900137 is OK
- 10.1007/3-540-27752-8 is OK
- 10.1101/gr.092759.109 is OK
- 10.3389/fnhum.2014.00448 is OK
- 10.1007/bf00198091 is OK
- 10.1007/pl00007990 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published

. Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1353

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1353, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by Kevin-Mattheus-Moerman** [comment]
I read the paper and it looks good

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1235977947573555201) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1355
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01982](https://doi.org/10.21105/joss.01982)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by Kevin-Mattheus-Moerman** [comment]
@openjournals/dev the DOI resolves but I get a 404 on the PDF link. Can you see what is happening?

**Comment by arfon** [comment]
@Kevin-Mattheus-Moerman it's working for me now. Sometimes there's a delay in the GitHub Pages build which can result in a 404 for a few minutes.

<img width="1291" alt="Screen Shot 2020-03-06 at 4 45 19 PM" src="https://user-images.githubusercontent.com/4483/76125039-f8798480-5fc9-11ea-8110-13abe224e33a.png">

**Comment by Kevin-Mattheus-Moerman** [comment]
@arfon ah right. It is all good now.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.01982/status.svg)](https://doi.org/10.21105/joss.01982)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01982">
  <img src="https://joss.theoj.org/papers/10.21105/joss.01982/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.01982/status.svg
   :target: https://doi.org/10.21105/joss.01982
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.01982/status.svg)](https://doi.org/10.21105/joss.01982)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by Kevin-Mattheus-Moerman** [comment]
@openjournals/dev can you help with this issue https://github.com/openjournals/joss/issues/700 highlighted by the author?

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.01982/status.svg)](https://doi.org/10.21105/joss.01982)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01982">
  <img src="https://joss.theoj.org/papers/10.21105/joss.01982/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.01982/status.svg
   :target: https://doi.org/10.21105/joss.01982
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.01982/status.svg)](https://doi.org/10.21105/joss.01982)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
